testscommon/chainSimulator/nodeHandlerMock.go

```diff
@@ -113,6 +116,13 @@ func (mock *NodeHandlerMock) SetStateForAddress(address []byte, state *dtos.Addr
 	return nil
 }
 
+func (mock *NodeHandlerMock) GetIncomingHeaderHandler() process.IncomingHeaderSubscriber {
```

Comment:
Missing comment.
Why is this part of the Node instead of runType? This is not ok

Reply:
IncomingHeaderHandler is not yet in RunType. I can also get it from ProcessComponents